packages/11ty/_plugins/shortcodes/annoref.js

```diff
@@ -0,0 +1,28 @@
+const chalkFactory = require('~lib/chalk')
+const { oneLine } = require('~lib/common-tags')
+const logger = chalkFactory(`Shortcodes:Annoref`)
+/**
+ * Style wrapped `content` as "backmatter"
+ *
+ * @param      {String}  content  content between shortcode tags
+ *
+ * @return     {boolean}  A styled HTML <div> element with the content
+ */
+module.exports = function (eleventyConfig) {
+  const getAnnotation = eleventyConfig.getFilter('getAnnotation')
+  return ({ anno='', fig, region='', text='' }) => {
+    const annoIds = anno.split(',').map((id) => id.trim())
+    const annotationUrls = annoIds.flatMap((id) => {
+      const annotation = getAnnotation(fig, id)
+      return annotation ? annotation.url : []
+    })
```

Comment:
Unless `annotation.url` is already an Array the `return` value here be `[annotation.url]` or `[]`, edited:
```suggestion
    const annotationUrls = anno.split(',').flatMap((id) => {
      const annotation = getAnnotation(fig, id.trim())
      return annotation ? [annotation.url] : []
    })
```

Reply:
I think they are equivalent, unless i'm missing a case?
```[1, null, 'gradoo'].flatMap((item) => item ? item : []) // [1, 'gradoo']```
```[1, null, 'gradoo'].flatMap((item) => item ? [item] : []) // [1, 'gradoo']```

Reply:
Indeed, I think they are equivalent; moving `trim()` into the flatMap function may still sense?